app/models/role_store.rb

```diff
@@ -10,7 +10,7 @@ def build_role_store_data
     josh_elder_email  = "JoshElder@northwestern.edu"
 
     Role.find_each do |role|
-      role_user_data = r.users.pluck(:username, :email).to_h
+      role_user_data = role.users.pluck(:username, :email).to_h
```

Comment:
Yes, I do think it would have been better to identify Josh's email in the script. Then we would not have needed any changes here. 
```
# lib/scripts/repo_export_in_batches.rb:28
role_store = RoleStore.new.build_role_store_data
role_store.data.each do |k,v| 
  if v["jbe2215"]
    v["jbe2215"] = "JoshElder@northwestern.edu"
  end
end
``` 
(☝️ needs testing)

Even better, handling it on creation. Let's open `#build_role_store_data` to accept arguments. Then in the script we can pass in Josh Elder's info to the method, while allowing more users to be added later. 
```
# app/models/role_store.rb
def build_role_store_data(user_new_emails=[])
  Role.find_each do |role|
    role_user_data = role.users.pluck(:username, :email).to_h

    user_new_emails.each do |user_email|
      if role_user_data[user_email[0]]
        role_user_data[user_email[0]] = user_email[1]
      end
    end

    @data[role.name] = role_user_data
  end
end
# Just my implementation. I am sure they are ways to improve
# the logic after accepting arguments

#  lib/scripts/repo_export_in_batches.rb
user_new_emails = [
  ["jbe2215", "JoshElder@northwestern.edu"], 
  ["abc123", "JohnSmith@email.com"]
]
role_store = RoleStore.new.build_role_store_data(user_new_emails)
```
(☝️ also needs testing, and easier to write tests for)

Sorry for the long and belated suggestion. I am okay with the current fix, but I saw something that I thought would be worth pointing out. Mark this for a future downtime refactor, or when another user needs an email to be corrected.